src/peft/utils/save_and_load.py

```diff
@@ -198,6 +198,8 @@ def load_peft_weights(model_id: str, device: Optional[str] = None, **hf_hub_down
             filename=SAFETENSORS_WEIGHTS_NAME,
             revision=hf_hub_download_kwargs.get("revision", None),
             repo_type=hf_hub_download_kwargs.get("repo_type", None),
+            token=hf_hub_download_kwargs.get("token", None),
+            use_auth_token=hf_hub_download_kwargs.get("use_auth_token", None),
```

Comment:
I would add this just before the `file_exists` call and then do `token=token` in the call.
```py
token = hf_hub_download_kwargs.get("token", None)
if token is None:
    token = hf_hub_download_kwargs.get("use_auth_token", None)
```

`use_auth_token` is deprecated in `huggingface_hub` and we're trying to get rid of it in every library (in favor of `token`). If you pass it as done at the moment it will not fail -only print a warning- but the implementation in peft will not be future-proof.

Reply:
(note: this is not equivalent to `token = hf_hub_download_kwargs.get("token", None) or hf_hub_download_kwargs.get("use_auth_token", None)` because of the corner case where user passes `token=False` ti explicitly not send a token)